Create separate markdown page for local develpment (#172)
linode/provider-ceph: docs/DEVELOPMENT.md

# Development

Refer to Crossplane's [CONTRIBUTING.md] file for more information on how the
Crossplane community prefers to work. The [Provider Development][provider-dev]
guide may also be of use.

[CONTRIBUTING.md]: https://github.com/crossplane/crossplane/blob/master/CONTRIBUTING.md

> ## Running locally

Reviewer: ```suggestion
## Running Locally
```

Spin up the test environment, but with `provider-ceph` running locally in your terminal:

```
make dev
```

After you've made some changes, kill (Ctrl+C) the existing `provider-ceph` and re-run it:

```
make run
```

> ### Webhook support

Reviewer: ```suggestion
### Webhook Support
```

> Running the validation webhook locally is a bit tricky, but working out of the box.

Reviewer: ```suggestion
Running the validation webhook locally is a bit tricky, but it works out of the box.
```

Firt of all cluster provisioner script changes `ValidatingWebhookConfiguration`, to point to a
[localtunnel](https://github.com/localtunnel/localtunnel) instance (created by the script).
This endpoint has a valid TLS certification aprooved by Kubernetes, so validation requests should be served by the local process.

> ## Debugging locally

Reviewer: ```suggestion
## Debugging Locally
```

Spin up the test environment, but with `provider-ceph` running locally in your terminal:

```
make mirrord.cluster mirrord.run
```

For debugging please install `mirrord` plugin in your IDE of choice.

> ### Webhook support

Reviewer: ```suggestion
### Webhook Support
```
